region/wreckedship/main/Bowling Alley.json

```diff
@@ -1003,10 +1014,55 @@
             "h_canShineChargeMaxRunway"
           ]}
         ]},
-        {"shinespark": {"frames": 40}}
+        {"or": [
+          {"shinespark": {"frames": 40, "excessFrames": 3}},
+          {"and": [
+            "canShinechargeMovement",
+            "canMidairShinespark",
+            {"shinespark": {"frames": 32, "excessFrames": 2}}
+          ]},
+          {"and": [
+            "canShinechargeMovementComplex",
```

Comment:
With wall jumps my best was 10 shinespark frames but getting past the spikes was extremely annoying.
With hijump I made it to the top before sparking.